feat: inline assets loading (#160)
callstack/repack: website/docs/configuration/guides/inline-assets.md

# Inline Assets

> By default, Re.Pack's [Assets loader](/docs/configuration/loaders/assets-loader) is configured to allow you to use assets in a normal way.

Reviewer: ```suggestion
By default, Re.Pack's [Assets loader](/docs/configuration/loaders/assets-loader) is configured to allow to import an asset and output it as as static file in `assets/` or `drawable-*` directories, similarly to Metro.
```

> By default, Re.Pack's [Assets loader](/docs/configuration/loaders/assets-loader) is configured to allow you to use assets in a normal way.

Reviewer: Use relative path in links

> This is not the case when it comes to the remote chunks that use assets which are not available in the main app. In this scenario remote chunk must have `inlined assets`.

Reviewer: ```suggestion
In some cases, you might want to change that behaviour and inline the assets into the JavaScript bundle. 

Common examples of such use case are:
- Out-of-tree platforms that don't support static assets in a similar way as Metro or don't support static assets at all.
- [Code splitting](../../code-splitting/usage) with static assets used by [remote chunks](../../code-splitting/glossary#remote-chunks)
```

> ## Pre-requisites

Reviewer: ```suggestion
## Usage
```

To inline assets you have to pass `inline` option to the [Assets loader](/docs/configuration/loaders/assets-loader).

As a result we will get assets in the form of:

```js
// remote.chunk.bundle.js
// ...
module.exports = { uri: 'base64string' };
```

> _Asset will be available in the chunk because it has been embedded into it._

Reviewer: ```suggestion
:::info

If you're using [Code Splitting](#) or [Module Federation](#), all assets inlined into remote chunks or containers will be available to the host application and should be properly rendered.

:::
```
Please replace `#` with correct relative paths
